pkg/fakerp/middleware.go

```diff
@@ -0,0 +1,30 @@
+package fakerp
+
+import (
+	"net/http"
+)
+
+func (s *Server) logger(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		s.log.Debugf("%s %s %s", r.RemoteAddr, r.Method, r.URL)
+		handler.ServeHTTP(w, r)
+	})
+}
+
+func (s *Server) validator(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Shouldn't you invalidate anything other than GET, PUT, and DELETE here too?

Reply:
The router takes care of that. A mapping for `s.router.Delete("/foo", s.handleFoo)` allows the router to only handle DELETES for /foo and returns a `405 Method not allowed` for all other methods

*Start fake rp*
```
charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ go run cmd/fakerp/main.go 
INFO[2019-01-11T10:41:36+01:00] starting the fake resource provider          
INFO[2019-01-11T10:41:36+01:00] starting server on localhost:8080            
```
*Test*
```
charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ curl -I -XGET http://localhost:8080/foo
HTTP/1.1 405 Method Not Allowed
Date: Fri, 11 Jan 2019 09:41:56 GMT
Content-Length: 0

charles@titan:~/Documents/go/src/github.com/openshift/openshift-azure$ curl -I -XDELETE http://localhost:8080/foo
HTTP/1.1 200 OK
Date: Fri, 11 Jan 2019 09:42:07 GMT
Content-Length: 0

```

*Logs*
```      
DEBU[2019-01-11T10:41:56+01:00] 127.0.0.1:47362 GET /foo (curl/7.58.0)       
DEBU[2019-01-11T10:42:07+01:00] 127.0.0.1:47364 DELETE /foo (curl/7.58.0) 
```